prow/repoowners/repoowners.go

```diff
@@ -409,12 +400,35 @@ func (o *RepoOwners) walkFunc(path string, info os.FileInfo, err error) error {
 	return nil
 }
 
+// ParseOwners will open and unmarshal OWNERS into either a SimpleConfig or a FullConfig
+// Returns an error if the file cannot be read or unmarshalled
+func ParseOwners(path string) (SimpleConfig, FullConfig, error) {
```

Comment:
Can we figure out a cleaner interface here? Returning three optionally nil objects and asking the caller to determine which ended up being valid is not the clearest

Reply:
Do we ever really need to load _either_? Let's just have a load func for each config type